sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -275,8 +276,37 @@ object LogicalPlanIntegrity {
    * Some plan transformers (e.g., `RemoveNoopOperators`) rewrite logical
    * plans based on this assumption.
    */
-  def checkIfExprIdsAreGloballyUnique(plan: LogicalPlan): Boolean = {
-    checkIfSameExprIdNotReused(plan) && hasUniqueExprIdsForOutput(plan)
+  def validateExprIdUniqueness(plan: LogicalPlan): Unit = {
+    if (!LogicalPlanIntegrity.checkIfSameExprIdNotReused(plan)) {
+      throw SparkException.internalError("Cannot reuse the exprId in Alias")
+    }
+    if (!LogicalPlanIntegrity.hasUniqueExprIdsForOutput(plan)) {
+      throw SparkException.internalError(
+        "Some Attributes have the same exprId but different data types")
```

Comment:
```suggestion
        "Some Attributes have the same exprId but different data types")
```
```suggestion
        "Multiple attributes have the same expression ID but different data types")
```

also can this mention the names, data types, and expression IDs of the attributes to help with future potential debugging?